utils/FactoryDeployerHelper.ts

```diff
@@ -63,7 +55,49 @@ export class FactoryDeployerHelper {
             // abi field is not used but is a required arg. Empty abi would be fine
             abi: (contractFactory as any).constructor.abi,
             address: contractAddress,
+            bytecode: bytecode,
+            args: constructorArgs,
         });
         return contractAddress;
     }
+
+    private async init() {
+        if (this.genericFactory) {
+            // Already initialized.
+            return;
+        }
+        const factoryConfig: FactoryConfig =
+            !config.isNativeChain() && hre.network.name.includes('hardhat')
+                ? factoryShanghaiJson
+                : factoryJson;
+        this.genericFactory = GenericFactory__factory.connect(factoryConfig.address, this.owner);
+        if ((await this.owner.provider!.getCode(factoryConfig.address)) !== '0x') {
```

Comment:
Maybe use `ethers.provider` for `getCode` & `broadcastTransaction` to harmonize with what we generally have

Reply:
Agree [Use ethers.provider instead of signer.provider](https://github.com/iExecBlockchainComputing/PoCo/pull/206/commits/8cd2d7c8755fb76c88ea8ca5e93f190c681fb4ed)